### PR TITLE
[bugFix][Connector-V2][Doris] The multi-FE configuration is supported

### LINKED ISSUE #6337: 2.3.4同步数据到doris异常
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

读取数据同步至doris报错，CheckpointCoordinator inside have error.
从2.3.3升级到2.3.4，同样的脚本运行不了

### SeaTunnel Version

2.3.4

### SeaTunnel Config

```conf
env {
    job.name = "douyin_xingtu_star_works_record"
    job.mode = "BATCH"
}

source {
    Jdbc {
    url = "jdbc:mysql://ccccc:3392/data_center?zeroDateTimeBehavior=convertToNull"
    driver = "com.mysql.cj.jdbc.Driver"
    user = ""
    password = ""
    query = "select * from  douyin_xingtu_star_works_record_v3"
    result_table_name ="source_kdouyin_xingtu_star_works_record_v3"
    }
}
transform {
}
sink {
      Doris {
        source_table_name="source_kdouyin_xingtu_star_works_record_v3"
        fenodes = "xxx:8030,xxx:8030,xxxx:8030"
        username = ""
        password = ""
        database = "ims_wemedia"
        table = "douyin_video_xingtu"
        sink.enable-2pc = "true"
        sink.label-prefix = "label_douyin_video_xingtu_20240204"
        doris.config = {
          format="json"
          read_json_by_line="true"
        }
    }
}
```

### Running Command

```shell
/bin/seatunnel.sh --config douyin_xingtu_video_to_doris.conf -e local
```

### Error Exception

```log
2024-02-04 17:38:39,953 ERROR [o.a.s.c.s.SeaTunnel           ] [main] - Exception StackTrace:org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:199)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException: org.apache.seatunnel.engine.server.checkpoint.CheckpointException: CheckpointCoordinator inside have error.
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointCoordinator.handleCoordinatorError(CheckpointCoordinator.java:270)
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointCoordinator.handleCoordinatorError(CheckpointCoordinator.java:266)
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointCoordinator.reportCheckpointErrorFromTask(CheckpointCoordinator.java:372)
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointManager.reportCheckpointErrorFromTask(CheckpointManager.java:183)
        at org.apache.seatunnel.engine.server.checkpoint.operation.CheckpointErrorReportOperation.run(CheckpointErrorReportOperation.java:48)
        at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:175)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:139)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123)
        at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: org.apache.seatunnel.connectors.doris.exception.DorisConnectorException: ErrorCode:[Doris-01], ErrorDescription:[stream load error]
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisSinkWriter.initializeLoad(DorisSinkWriter.java:115)
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisSinkWriter.<init>(DorisSinkWriter.java:98)
        at org.apache.seatunnel.connectors.doris.sink.DorisSink.createWriter(DorisSink.java:83)
        at org.apache.seatunnel.connectors.seatunnel.common.multitablesink.MultiTableSink.createWriter(MultiTableSink.java:70)
        at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.restoreState(SinkFlowLifeCycle.java:297)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.lambda$restoreState$16(SeaTunnelTask.java:425)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.restoreState(SeaTunnelTask.java:422)
        at org.apache.seatunnel.engine.server.checkpoint.operation.NotifyTaskRestoreOperation.lambda$null$0(NotifyTaskRestoreOperation.java:106)
        at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.http.client.ClientProtocolException: URI does not specify a valid host name: http://172.18.222.76:8030,172.18.222.74:8030/api/ims_wemedia/douyin_video_xingtu/_stream_load
        at org.apache.http.impl.client.CloseableHttpClient.determineTarget(CloseableHttpClient.java:95)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisStreamLoad.abortPreCommit(DorisStreamLoad.java:144)
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisSinkWriter.initializeLoad(DorisSinkWriter.java:112)
        ... 21 more

        ... 11 more

        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:191)
        ... 2 more

2024-02-04 17:38:39,954 ERROR [o.a.s.c.s.SeaTunnel           ] [main] -
===============================================================================



Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:199)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException: org.apache.seatunnel.engine.server.checkpoint.CheckpointException: CheckpointCoordinator inside have error.
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointCoordinator.handleCoordinatorError(CheckpointCoordinator.java:270)
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointCoordinator.handleCoordinatorError(CheckpointCoordinator.java:266)
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointCoordinator.reportCheckpointErrorFromTask(CheckpointCoordinator.java:372)
        at org.apache.seatunnel.engine.server.checkpoint.CheckpointManager.reportCheckpointErrorFromTask(CheckpointManager.java:183)
        at org.apache.seatunnel.engine.server.checkpoint.operation.CheckpointErrorReportOperation.run(CheckpointErrorReportOperation.java:48)
        at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248)
        at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:175)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:139)
        at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123)
        at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: org.apache.seatunnel.connectors.doris.exception.DorisConnectorException: ErrorCode:[Doris-01], ErrorDescription:[stream load error]
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisSinkWriter.initializeLoad(DorisSinkWriter.java:115)
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisSinkWriter.<init>(DorisSinkWriter.java:98)
        at org.apache.seatunnel.connectors.doris.sink.DorisSink.createWriter(DorisSink.java:83)
        at org.apache.seatunnel.connectors.seatunnel.common.multitablesink.MultiTableSink.createWriter(MultiTableSink.java:70)
        at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.restoreState(SinkFlowLifeCycle.java:297)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.lambda$restoreState$16(SeaTunnelTask.java:425)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.restoreState(SeaTunnelTask.java:422)
        at org.apache.seatunnel.engine.server.checkpoint.operation.NotifyTaskRestoreOperation.lambda$null$0(NotifyTaskRestoreOperation.java:106)
        at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1626)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.http.client.ClientProtocolException: URI does not specify a valid host name: http://172.18.222.76:8030,172.18.222.74:8030/api/ims_wemedia/douyin_video_xingtu/_stream_load
        at org.apache.http.impl.client.CloseableHttpClient.determineTarget(CloseableHttpClient.java:95)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisStreamLoad.abortPreCommit(DorisStreamLoad.java:144)
        at org.apache.seatunnel.connectors.doris.sink.writer.DorisSinkWriter.initializeLoad(DorisSinkWriter.java:112)
        ... 21 more

        ... 11 more

        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:191)
        ... 2 more
2024-02-04 17:38:39,958 INFO  [s.c.s.s.c.ClientExecuteCommand] [ForkJoinPool.commonPool-worker-57] - run shutdown hook because get close signal
```

### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

1.8.0_181

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
### Purpose of this pull request
close #6337 

### How was this patch tested?
Current test

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Could you add a test case for this?

**Comment by Carl-Zhou-CN** [comment]
> Could you add a test case for this?

Ok, I'll add it, but right now I can't find a multi-Fe docker image

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java [hunk 1]
```diff
@@ -237,7 +237,8 @@ static String[] parseIdentifier(String tableIdentifier, Logger logger)
     }
 
     @VisibleForTesting
-    static String randomEndpoint(String feNodes, Logger logger) throws DorisConnectorException {
+    public static String randomEndpoint(String feNodes, Logger logger)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java:240]: Use UT to test this method is enough.

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java:240]: done @Hisoka-X

File: seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/rest/models/RestServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        Assertions.assertTrue(
+                list.contains(RestService.randomEndpoint("fe_host1:fe_http_port1", log)));
+        Assertions.assertTrue(
+                list.contains(
+                        RestService.randomEndpoint(
+                                "fe_host1:fe_http_port1,fe_host2:fe_http_port2", log)));
+        Assertions.assertTrue(
+                list.contains(
+                        RestService.randomEndpoint(
+                                "fe_host2:fe_http_port2,fe_host1:fe_http_port1", log)));
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/rest/models/RestServiceTest.java]: This test case can not prove the endpoint is random. Please refer https://github.com/apache/seatunnel/blob/df1954d52076c7ba6d3d764c9e4e58725193a102/seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/resourcemanager/ResourceManagerTest.java#L86

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/rest/models/RestServiceTest.java]: done

**Comment by Carl-Zhou-CN** [comment]
@Hisoka-X cc

File: seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/rest/models/RestServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        List<String> list =
+                Arrays.asList(
+                        "fe_host1:fe_http_port1",
+                        "fe_host2:fe_http_port2",
+                        "fe_host3:fe_http_port3",
+                        "fe_host4:fe_http_port4",
+                        "fe_host5:fe_http_port5");
+
+        boolean hasDifferentWorker = false;
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/rest/models/RestServiceTest.java]:
```suggestion
        boolean hasDifferentAddress = false;
```

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-doris/src/test/java/org/apache/seatunnel/connectors/doris/rest/models/RestServiceTest.java]: done

**Review by Hisoka-X — APPROVED** [review]
LGTM if ci passed.

**Review by hailin0 — APPROVED** [review]

**Pull request merged**